### PR TITLE
Add check of event support

### PR DESCRIPTION
**Pull request by kelvin-cao**
Not all events are supported by every gen/variant of the Switchtec
firmware. To solve this, since Gen4, a new bit in each event header
is introduced to indicate if an event is supported by the firmware.

The bit is set by the firwmare and read by the software.

### CODE REVIEW

File: switchtec.c [hunk 1]
```diff
@@ -1004,6 +1020,9 @@ static int ioctl_event_ctl(struct switchtec_dev *stdev,
 	if (ctl.flags & SWITCHTEC_IOCTL_EVENT_FLAG_UNUSED)
 		return -EINVAL;
 
+	if (!event_supported(stdev, ctl.event_id, ctl.index))
```
Comment (lsgunth) [inline switchtec.c]: This definitely seems like the wrong place for this. It ignores EVENT_IDX_ALL and it's an extra read that's not needed. Just move the check into event_ctl() where it already reads the hdr.

You'll probably need to also skip the specific error when doing all events.

Comment (kelvin-cao) [reply switchtec.c]: > You'll probably need to also skip the specific error when doing all events.

Sorry, I didn't get your point.

Comment (lsgunth) [reply switchtec.c]: My point was if the user is doing an IDX_ALL ioctl, you don't want a single not supported event to stop the loop.

**Comment by kelvin-cao** [comment]
Updated, thanks.

**Pull request merged**